.github/workflows/ci-backend.yml

```diff
@@ -245,6 +257,9 @@ jobs:
               with:
                   python-version: 3.8
 
+            - name: Install SAML (python3-saml) dependencies
+              run: sudo apt-get install libxml2-dev libxmlsec1-dev libxmlsec1-openssl
```

Comment:
As these OS-level deps are now required for SAML to run properly, I was considering creating a separate requirements file for EE and a Dockerfile for FOSS. That way if you're on FOSS you won't have to install the additional deps you'll never use. But might be too much complexity for the benefit we're actually getting. Thoughts? @fuziontech @tiina303 @macobo

Reply:
Hm, these deps are rather annoying, but as far as I can google: 1) there's no way around `python3-saml` if we want SAML in python, as it's the only real feature complete library on the topic, 2) these external deps are thus unavoidable.

I don't think the complexity in maintaining two separate Dockerfiles is worth it.

However I'm worried we're probably forgetting a few places where we would need to run this (cloud? plugin server tests? some obscure installation method? not sure). Thus is there any way for SAML to fail in a friendly way if any of these deps are missing?

Reply:
+1 to lets try to avoid the complexity of two separate Dockerfiles/setups.

Reply:
Ooof. Just saw we're doing SAML! There's a few more issues that crop up (I tried doing this in Python a while ago): The OS deps + python3-saml don't work well with > python3.6 (this might have changed since then - but worth confirming in tests :).

IIRC, libxmlsec1-dev also had some install issues on Mac, and specific Linux versions, which implies we'd better adapt a docker workflow for working with them. (using the correct base images)

Reply:
Thanks everyone for the feedback!
- Agreed @neilkakkar, after testing this it looks like it works properly with our relevant Docker configs.
- Perfect, will keep single file config and this will get shipped with FOSS even if it's not needed.
- Good catch @mariusandra! Will open PRs in the relevant repos.
- As far as I could see, there's one option for SAML deps to fail silently, but I'm not sure if it's too much and will instead cause maintenance issues (see below). **Looking for feedback on this.**


1. Remove `python3-saml` dependency from `requirements.in` (and related compilation on `requirements.txt`)
2. Run `pip install python3-saml==1.12.0` separately in all the workflows

Two problems with the approach above:
- Ton of places to maintain this dependency installation (easy to miss too)
- Lose the benefits of `pip-compile` to try to reduce conflicting dependencies.

Reply:
Yeah, the SAML failing silently option doesn't make sense here. At the end of the day, we shouldn't be afraid to ship the software or libraries we need (unless it's something under GPL :D), so just go for it.

I tried `pip`pin' locally, and got:

```bash
  Building wheel for xmlsec (PEP 517) ... error
  ERROR: Command errored out with exit status 1:
   command: /Users/marius/Projects/PostHog/posthog/env/bin/python3 /Users/marius/Projects/PostHog/posthog/env/lib/python3.9/site-packages/pip/_vendor/pep517/_in_process.py build_wheel /var/folders/_d/sftx4p791gv85n0k1bwl_06m0000gn/T/tmpqmjyhzui
       cwd: /private/var/folders/_d/sftx4p791gv85n0k1bwl_06m0000gn/T/pip-install-9z74g3nv/xmlsec_0a030c9fa67d47628cc2c5d866d2e89a
  Complete output (20 lines):
  /private/var/folders/_d/sftx4p791gv85n0k1bwl_06m0000gn/T/pip-build-env-4kipbe0x/overlay/lib/python3.9/site-packages/setuptools/dist.py:697: UserWarning: Usage of dash-separated 'description-file' will not be supported in future versions. Please use the underscore name 'description_file' instead
    warnings.warn(
  /private/var/folders/_d/sftx4p791gv85n0k1bwl_06m0000gn/T/pip-build-env-4kipbe0x/overlay/lib/python3.9/site-packages/setuptools/dist.py:697: UserWarning: Usage of dash-separated 'build-requires' will not be supported in future versions. Please use the underscore name 'build_requires' instead
    warnings.warn(
  /private/var/folders/_d/sftx4p791gv85n0k1bwl_06m0000gn/T/pip-build-env-4kipbe0x/overlay/lib/python3.9/site-packages/setuptools/dist.py:697: UserWarning: Usage of dash-separated 'upload-dir' will not be supported in future versions. Please use the underscore name 'upload_dir' instead
    warnings.warn(
  running bdist_wheel
  running build
  running build_py
  package init file 'src/xmlsec/__init__.py' not found (or not a regular file)
  creating build
  creating build/lib.macosx-11-arm64-3.9
  creating build/lib.macosx-11-arm64-3.9/xmlsec
  copying src/xmlsec/py.typed -> build/lib.macosx-11-arm64-3.9/xmlsec
  copying src/xmlsec/tree.pyi -> build/lib.macosx-11-arm64-3.9/xmlsec
  copying src/xmlsec/__init__.pyi -> build/lib.macosx-11-arm64-3.9/xmlsec
  copying src/xmlsec/constants.pyi -> build/lib.macosx-11-arm64-3.9/xmlsec
  copying src/xmlsec/template.pyi -> build/lib.macosx-11-arm64-3.9/xmlsec
  running build_ext
  error: xmlsec1 is not installed or not in path.
  ----------------------------------------
  ERROR: Failed building wheel for xmlsec
Failed to build xmlsec
ERROR: Could not build wheels for xmlsec which use PEP 517 and cannot be installed directly
```

Running `brew install xmlsec1` did solve it for me though.